Assets/Plugins/Editor/Uplift/Common/Logger.cs

```diff
@@ -0,0 +1,305 @@
+using System;
+using UnityEngine;
+using Object = UnityEngine.Object;
+
+namespace Uplift.Common
+{
+	//
+	// Summary:
+	//     Class containing methods to ease debugging while developing a game.
+	public sealed class Logger
+	{
+		private bool verbose = false;
+		public bool IsVerbose { get { return verbose; } }
```

Comment:
a usual design would be to have the level at which you log

and then you don't need to have Log and VerboseLog methods.

Reply:
Yep, but I wanted to keep as close as Debug.Log as I could. So I would have needed to reimplement the LogType enum, which I didn't want. This is a simple mapping that also enables us to Verbose warnings for instance.

Reply:
Actually Unity supports this now. So we could just get rid of this implementation and reuse the internals.

https://docs.unity3d.com/ScriptReference/Logger-filterLogType.html

Probably by doing something like
```
Logger logger = new Logger(Debug.unityLogger.logHandler);
logger.filterLogType = LogType.Error;
```

```

Reply:
New implementation uses that!